spec/schema.spec.js

```diff
@@ -0,0 +1,77 @@
+"use strict";
+
+describe("schema", () => {
+    var fs, promiseMock, schema, tv4;
+
+    beforeEach(() => {
+        var nodeValidator;
+
+        fs = jasmine.createSpyObj("fs", [
+            "readdirAsync",
+            "readFileAsync"
+        ]);
+        fs.readFileAsync.andCallFake((fn) => {
+            if (fn.match("email.json")) {
+                return promiseMock.resolve(new Buffer('{"id": "email", "type": "string", "format": "email"}', "binary"));
+            }
+
+            if (fn.match("number.json")) {
+                return promiseMock.resolve(new Buffer('{"id": "number", "type": "number", "minimum": 5}', "binary"));
+            }
+
+            if (fn.match("email-no-id.json")) {
+                return promiseMock.resolve(new Buffer('{"type": "string", "format": "email"}', "binary"));
+            }
+
+            if (fn.match("email-parse-error.json")) {
+                return promiseMock.resolve(new Buffer('{"type: "string", "format": "email"}', "binary"));
+            }
+
+            return promiseMock.reject("Invalid file: " + fn.toString());
+        });
+        nodeValidator = require("validator");
+        promiseMock = require("./mock/promise-mock");
+        tv4 = require("tv4");
+        schema = require("../lib/schema")(fs, nodeValidator, promiseMock, tv4);
+    });
+    describe(".loadSchemaAsync()", () => {
+        beforeEach(() => {
+            tv4.addSchema = jasmine.createSpy("tv4.addSchema").andCallThrough();
+        });
+        it("loads a schema and validates against it", (done) => {
```

Comment:
This should be moved to a separate `describe()`.  I can see us having additional formats validated and the guts of this really is validating against the email format.